core/utils.py

```diff
@@ -0,0 +1,247 @@
+        if value is not None:
+            cleaned[field] = bool(value)
+    
+    # List fields - ensure they're lists
+    list_fields = [
+        'title', 'partners', 'contributor_type', 'packages_eic',
+        'packages_editor', 'packages_submitted', 'packages_reviewed'
+    ]
+    
+    for field in list_fields:
+        value = contributor.get(field, [])
+        if value is None:
+            cleaned[field] = []
+        elif isinstance(value, list):
+            # Filter out empty/None values and strip strings
+            cleaned[field] = [
+                str(item).strip() for item in value 
+                if item is not None and str(item).strip()
+            ]
+        else:
+            # Single value, convert to list
+            cleaned[field] = [str(value).strip()] if str(value).strip() else []
+    
+    return cleaned
+
+
+def get_recent_contributors(count: int = 4) -> List[Dict[str, Any]]:
+    """
+    Get the most recent contributors.
+    
+    Args:
+        count: Number of recent contributors to return.
+        
+    Returns:
+        List of recent contributor dictionaries, sorted by date_added descending.
+    """
+    try:
+        contributors = fetch_contributors_yaml()
+        
+        # Clean all contributor data
+        cleaned_contributors = []
+        for contributor in contributors:
+            try:
+                cleaned = clean_contributor_data(contributor)
+                cleaned_contributors.append(cleaned)
+            except ContributorDataError as e:
+                logger.warning(f"Skipping invalid contributor: {e}")
+                continue
+        
+        # Sort by date_added (newest first), then by sort field
+        def sort_key(contributor):
+            date_added = contributor.get('date_added')
+            sort_value = contributor.get('sort', 999999)  # Default high sort value
+            
+            # Contributors without dates go to the end
+            if date_added is None:
+                return (datetime.min.date(), sort_value)
+            
+            return (date_added, sort_value)
+        
+        sorted_contributors = sorted(
+            cleaned_contributors, 
+            key=sort_key, 
+            reverse=True
+        )
+        
+        return sorted_contributors[:count]
+        
+    except ContributorDataError as e:
+        logger.error(f"Failed to get recent contributors: {e}")
+        return []
+    except Exception as e:
+        logger.error(f"Unexpected error getting recent contributors: {e}")
+        return []
+
+
+def generate_github_avatar_url(github_image_id: int) -> str:
+    """
+    Generate GitHub avatar URL from image ID.
+    
+    Args:
+        github_image_id: GitHub user's image ID.
+        
+    Returns:
+        GitHub avatar URL.
+    """
+    return f"https://avatars.githubusercontent.com/u/{github_image_id}?s=400&v=4"
+
+
+def generate_github_profile_url(github_username: str) -> str:
```

Comment:
@Phinart98 we use [numpy style docstrings](https://numpydoc.readthedocs.io/en/latest/format.html#sections) as a convention here in pyOpenSci

if you look at pyosMeta code you will see how it's applied there. please implement that here as well.